src/main/java/mate/academy/dao/BookDao.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.dao;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.repository.BookDaoRepository;
+import mate.academy.utility.Connector;
+
+@Dao
+public class BookDao implements BookDaoRepository {
+    private final Connector connector;
+
+    public BookDao() {
+        connector = new Connector();
+    }
+
+    public Book create(Book book) {
+        String sqlInsert = "INSERT INTO books (title, price) VALUES (?,?);";
+        try (PreparedStatement preparedStatement =
+                     Connector.getConnection().prepareStatement(sqlInsert)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "There is a problem with connection to DB in create method!!", e);
+        }
+        return book;
+    }
+
+    public Optional<Book> findById(Long id) {
+        String sqlFindById = "SELECT id, title, price FROM books WHERE id = ?;";
+        Book book = new Book();
+        try (PreparedStatement preparedStatement =
+                     Connector.getConnection().prepareStatement(sqlFindById)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+                book.setTitle(resultSet.getString(2));
+                book.setPrice(resultSet.getBigDecimal(3));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "There is a problem with connection to DB in findById method!!", e);
+        }
+        return Optional.of(book);
+    }
+
+    public List<Book> findAll() {
+        String sqlGetAll = "SELECT * FROM books;";
+        List<Book> books;
+        try (PreparedStatement statement = Connector.getConnection().prepareStatement(sqlGetAll)) {
```

Comment:
```suggestion
        try (ConnectionUtil connection = ConnectionUtil.getConnection(); 
                             PreparedStatement statement = connection.prepareStatement(sqlGetAll)) {
```
you should close connections, to achieve that we have to list connection as a separate parameter in try-with-resources